### PR TITLE
refactor: make most python param names mandatory to be specified

### PR DESCRIPTION
**Pull request by Sparkier**
# Description

fix ZEN-203

### CODE REVIEW

File: zeno_client/client.py [hunk 1]
```diff
@@ -154,7 +155,7 @@ def upload_dataset(
         print("Successfully uploaded data")
 
     def upload_system(
-        self, name: str, df: pd.DataFrame, id_column: str, output_column: str
+        self, name: str, df: pd.DataFrame, *, id_column: str, output_column: str
```
Comment (cabreraalex) [inline zeno_client/client.py:158]: Should we swap these? 

And/or should we check that they are the right types in the body? AFAIK types are only suggestions, don't actually error if you pass the wrong type right?

Comment (cabreraalex) [reply zeno_client/client.py:158]: sorry mean swapping name and df to be consistent with upload dataset. IDK what is more natural.... maybe this is fine?

File: zeno_client/client.py [hunk 2]
```diff
@@ -263,6 +264,7 @@ def __init__(self, api_key, endpoint=DEFAULT_BACKEND) -> None:
     def create_project(
         self,
         name: str,
+        *,
```
Comment (cabreraalex) [inline zeno_client/client.py:267]: Should view be named? Guessing yes but idk

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: zeno_client/client.py (hunk 1 above)
Comment (Sparkier) [reply zeno_client/client.py:158]: No strong preference either, can swap and maybe make name named as well?

Hmm could check but we do nowhere else, right? Or which types would you check?

Comment (cabreraalex) [reply zeno_client/client.py:158]: yeah maybe swap and force named? I would just check that name is a string and that df is a dataframe. Maybe we should do checks across the board?

Comment (Sparkier) [reply zeno_client/client.py]: I swapped them and forced named names as well now just df can be unnamed. 

Type-checking: could do but kinda weird for a python library, no? If we check we should probably do throughout and then in a separate PR.

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**